diff -urN libvorbis-1.0beta4.org/lib/Makefile.am libvorbis-1.0beta4/lib/Makefile.am
--- libvorbis-1.0beta4.org/lib/Makefile.am	Sun Apr 29 00:04:26 2001
+++ libvorbis-1.0beta4/lib/Makefile.am	Sun Apr 29 01:10:55 2001
@@ -30,7 +30,7 @@
 barkmel_SOURCES = barkmel.c
 tone_SOURCES = tone.c
 psytune_SOURCES = psytune.c
-psytune_LDFLAGS = -static
+psytune_LDFLAGS = 
 psytune_LDADD = libvorbis.la
 
 EXTRA_DIST = lookups.pl iir.c
--- libvorbis-1.1.0.orig/vq/Makefile.am	Sun Apr 29 00:04:26 2001
+++ libvorbis-1.1.0/vq/Makefile.am	Sun Apr 29 01:10:38 2001
@@ -9,7 +9,7 @@
 	latticetune huffbuild residuesplit distribution
 CLEANFILES = $(EXTRA_PROGRAMS)
 
-AM_LDFLAGS = -static
+AM_LDFLAGS = 
 LDADD = ../lib/libvorbis.la
 
 lspvqtrain_SOURCES = lspdata.c train.c vqgen.c bookutil.c\
