diff -urN libvorbis-1.2.2/configure.ac libvorbis-1.2.2.new/configure.ac
--- libvorbis-1.2.2/configure.ac	2009-06-24 17:23:47.000000000 +0200
+++ libvorbis-1.2.2.new/configure.ac	2009-07-01 22:02:18.620749535 +0200
@@ -109,9 +109,9 @@
 	case $host in 
 	*86-*-linux*)
 		DEBUG="-g -Wall -Wextra -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
-		CFLAGS="-O20 -ffast-math -mno-ieee-fp -D_REENTRANT -fsigned-char"
+		CFLAGS="$CFLAGS -ffast-math -mno-ieee-fp -D_REENTRANT -fsigned-char"
 #              	PROFILE="-Wall -Wextra -pg -g -O20 -ffast-math -D_REENTRANT -fsigned-char -fno-inline -static"
-		PROFILE="-Wall -Wextra -pg -g -O20 -ffast-math -mno-ieee-fp -D_REENTRANT -fsigned-char -fno-inline"
+		PROFILE="-Wall -Wextra -pg -g $CFLAGS -ffast-math -mno-ieee-fp -D_REENTRANT -fsigned-char -fno-inline"
 
 		# glibc < 2.1.3 has a serious FP bug in the math inline header
 		# that will cripple Vorbis.  Look to see if the magic FP stack
@@ -153,13 +153,13 @@
   		PROFILE=${PROFILE}" -D__NO_MATH_INLINES"
   		fi;;
 	powerpc-*-linux*)
-		DEBUG="-g -Wall -Wextra -D_REENTRANT -D__NO_MATH_INLINES"
-		CFLAGS="-O3 -Wall -Wextra -ffast-math -mfused-madd -mcpu=750 -D_REENTRANT"
-		PROFILE="-pg -g -O3 -ffast-math -mfused-madd -mcpu=750 -D_REENTRANT";;
+		DEBUG="-g -Wall -Wextra -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
+		CFLAGS="$CFLAGS -Wall -Wextra -mfused-madd -D_REENTRANT -fsigned-char"
+		PROFILE="-pg -g $CFLAGS -mfused-madd -D_REENTRANT -fsigned-char";;
 	*-*-linux*)
 		DEBUG="-g -Wall -Wextra -D_REENTRANT -D__NO_MATH_INLINES -fsigned-char"
-		CFLAGS="-O20 -Wall -Wextra -ffast-math -D_REENTRANT -fsigned-char"
-		PROFILE="-pg -g -O20 -ffast-math -D_REENTRANT -fsigned-char";;
+		CFLAGS="$CFLAGS -Wall -Wextra -ffast-math -D_REENTRANT -fsigned-char"
+		PROFILE="-pg -g $CFLAGS -ffast-math -D_REENTRANT -fsigned-char";;
 	sparc-sun-*)
 		sparc_cpu=""
 		AC_MSG_CHECKING([if gcc supports -mv8])
